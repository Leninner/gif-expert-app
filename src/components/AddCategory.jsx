import React, { useState } from 'react';
import { Button, TextField } from '@mui/material';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('Handle submit');

    if (inputValue.trim().length > 2) {
      setCategories(inputValue);
      setInputValue('');
    }
  };

  return (
    <form className='divito' onSubmit={handleSubmit}>
      <TextField label='Category' focused onChange={handleInputChange} value={inputValue} />
      <p>{inputValue}</p>
      <Button variant='contained' type='submit'>
        Add Category
      </Button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
