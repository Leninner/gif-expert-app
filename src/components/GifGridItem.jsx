import React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';

export const GifGridItem = ({ title, url }) => {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardMedia image={url} component='img' height='140' />
      <CardContent>
        <Typography gutterBottom variant='h5' component='h2'>
          {title}
        </Typography>
      </CardContent>
    </Card>
  );
};
