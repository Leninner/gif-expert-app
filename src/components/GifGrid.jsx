import React from 'react';
import { GifGridItem } from './GifGridItem';
import Grid from '@mui/material/Grid';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h1>{category}</h1>
      <Grid container spacing={{ xs: 1, md: 3 }} columns={{ xs: 2, sm: 8, md: 12 }} sx={{ flexGrow: 0 }}>
        {loading && <h2>Loading...</h2>}
        {images.map((img) => (
          <Grid item xs={2} sm={4} md={4} key={img.id}>
            <GifGridItem {...img} />
          </Grid>
        ))}
      </Grid>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
