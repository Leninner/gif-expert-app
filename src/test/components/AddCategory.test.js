import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el componente Add Category', () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe funcionar el onChange en el input', () => {
    const input = wrapper.find('ForwardRef(TextField)'); // con esto se busca el input con MUI
    const value = 'Hola Mundo';

    input.simulate('change', { target: { value } });
    const parent = wrapper.find('p');

    expect(parent.text().trim()).toBe(value);
  });

  test('No debe postear la informaciÃ³n al hacer submit', () => {
    const form = wrapper.find('form');

    form.simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
});
