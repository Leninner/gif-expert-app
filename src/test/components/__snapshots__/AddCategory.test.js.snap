// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Add Category Debe mostrarse correctamente 1`] = `
<form
  className="divito"
  onSubmit={[Function]}
>
  <ForwardRef(TextField)
    focused={true}
    label="Category"
    onChange={[Function]}
    value=""
  />
  <p />
  <ForwardRef(Button)
    type="submit"
    variant="contained"
  >
    Add Category
  </ForwardRef(Button)>
</form>
`;
